### DISCUSSION TITLE
How to configure `lsp.servers` content?

### DISCUSSION THREAD

**Discussion by HexSleeves** [MunifTanjim/nougat.nvim#75] [Q&A] [answered]
### Did you check the docs and existing issues?

- [X] I am using latest nougat.nvim
- [X] I have read all the nougat.nvim docs
- [X] I have searched the existing issues of nougat.nvim

### Problem

get_content of LSP config is currently hardcoded to a set function

### Solution

Allow for a configurable opts to be passed to allow overriding the default implementation

### Alternatives

An alternative would be created the `Item` then merging it with another table that has my custom function

```
local item = nut.lsp({
  config = {},
  prefix = "  ",
  hl = { fg = color.green },
  sep_right = sep.right_chevron_solid(true),
  on_click = function()
    vim.cmd("LspInfo")
  end,
})

stl:add_item(Utils.merge(item, { content = custom_content_fn }))
```

**Comment by MunifTanjim** [comment, marked as answer]
What are you trying to config?

Did you try the `config.content`? For example, check https://github.com/MunifTanjim/nougat.nvim/issues/49#issuecomment-1823920378

```lua
local lsp_servers = nut.lsp.servers.create({
  ctx = {
    content = {
      lua_ls = "LuaLS",
    },
    hl = {
      lua_ls = { fg = "cyan" },
    },
  },
  priority = 1,
  hl = { bg = "gray", fg = "white" },
  sep_left = sep.left_lower_triangle_solid(true),
  config = {
    content = function(client, item)
      return {
        content = item.ctx.content[client.name] or client.name,
        hl = item.ctx.hl[client.name],
      }
    end,
    sep = " ",
  },
  suffix = " ",
})
```

**Comment by MunifTanjim** [comment]
A more detailed example: https://github.com/MunifTanjim/nougat.nvim/issues/49#issuecomment-1825004239
